credentials/credentials.go

```diff
@@ -215,6 +215,21 @@ func RequestInfoFromContext(ctx context.Context) (ri RequestInfo, ok bool) {
 	return ri, ok
 }
 
+// NewRequestInfoContext creates a new context with the given RequestInfo
+// attached to it.
+//
+// This RequestInfo will be accessible via RequestInfoFromContext.
+//
+// Intended to be used from tests for PerRPCCredentials implementations (that
+// often need to check connection's SecurityLevel). Should not be used from
+// non-test code: the gRPC client already prepares a context with the correct
+// RequestInfo attached when calling PerRPCCredentials.GetRequestMetadata.
+//
+// This API is experimental.
+func NewRequestInfoContext(ctx context.Context, ri RequestInfo) context.Context {
+	return icredentials.NewRequestInfoContext(ctx, ri)
```

Comment:
Can we move the internal functions here instead of passing calls through?

Reply:
Done, but internal/credentials/credentials.go still remains since it still has `ClientHandshakeInfoFromContext` and `NewClientHandshakeInfoContext`. Removing it will require making `NewClientHandshakeInfoContext` public as well.

Do you want me to do this? 

Strictly speaking it may also be useful for testing PerRPCCredentials.

Reply:
I would be fine with this, too, if you are interested in doing it.  The same caveats would apply: it should only be used for testing.  But if you don't need it, then nobody else has asked, so I'm fine with waiting until someone needs it, instead.

Reply:
I looked into how `ClientHandshakeInfoFromContext` is used (by actual prod code) and it is only used by xDS credentials implementation. I think for xds-related tests outside of grpc package (if any) to be useful, they will need access to other xDS internal parts (internal/xds package). For that reason I think it makes little sense to make it public now (also no one really needs it public).